src/main/js/tansaction/transaction-modal.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Bank account modal component.
+ *
+ * @author Xiaobing Hou
+ */
+import React from "react";
+import { Modal } from "react-bootstrap";
+import TransactionForm from "./transaction-form";
+
+const TransactionModal = ({
+  bankAccountId,
+  isAddTransaction,
```

Comment:
I saw this in a previous file and commented how you can avoid passing a flag down.
Example.
```JavaScript
selectedTransaction.counterparty || ""
```
Let's remove the flags you have been passing down.

Reply:
I tried this before, but got an error during the page load, so I changed it to this